lib/src/rules/use_string_buffer.dart

```diff
@@ -0,0 +1,212 @@
+// Copyright (c) 2017, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+library linter.src.rules.use_string_buffer;
+
+import 'package:analyzer/dart/ast/ast.dart';
+import 'package:analyzer/dart/ast/token.dart';
+import 'package:analyzer/dart/ast/visitor.dart';
+import 'package:analyzer/dart/element/element.dart';
+import 'package:linter/src/analyzer.dart';
+import 'package:linter/src/util/dart_type_utilities.dart';
+
+const _desc = r'Use string buffer to compose strings.';
+
+const _details = r'''
+
+**DO** use string buffer to compose strings.
```

Comment:
I think this rule (rightly) only lints for uses related to local variables (ignores fields and top-level variables). If that's true, it should probably be documented somewhere.